src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IForEachLoopStatement.vb

```diff
@@ -800,10 +861,80 @@ IForEachLoopStatement (Iteration variable: null) (LoopKind.ForEach) (OperationKi
         IfTrue: IBlockStatement (1 statements) (OperationKind.BlockStatement) (Syntax: 'If o IsNot  ... Return True')
             IReturnStatement (OperationKind.ReturnStatement) (Syntax: 'Return True')
               ReturnedValue: ILiteralExpression (OperationKind.LiteralExpression, Type: System.Boolean, Constant: True) (Syntax: 'True')
-        IfFalse: null]]>.Value
+        IfFalse: null
+  NextVariables(0)
+]]>.Value
 
             VerifyOperationTreeForTest(Of ForEachBlockSyntax)(source, expectedOperationTree)
         End Sub
 
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact(), WorkItem(17602, "https://github.com/dotnet/roslyn/issues/17602")>
+        Public Sub IForEachLoopStatement_FieldAsIterationVariable()
```

Comment:
>IForEachLoopStatement_FieldAsIterationVariable [](start = 19, length = 46)

Are there tests that use convert collection to generic IEnumerable, or use For Each pattern? Is for each over  a string covered? Same question for C# tests.

Reply:
Filed https://github.com/dotnet/roslyn/issues/22030, I will avoid adding more tests in this PR to scope it to API refactoring.